integrated_channel_plugin_sapsf/models.py

```diff
@@ -0,0 +1,135 @@
+"""
+Database models for Enterprise Integrated Channel SAP SuccessFactors.
+"""
+
+from __future__ import unicode_literals
+
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+from model_utils.models import TimeStampedModel, TimeFramedModel
+
+
+# Create your models here.
+
+@python_2_unicode_compatible
+class OCNWebServicesConfiguration(TimeStampedModel):
+    """
+    The Global configuration for integrating with SuccessFactors.
+    """
+
+    completion_status_api_path = models.CharField(max_length=255)
+    course_api_path = models.CharField(max_length=255)
+    oauth_api_path = models.CharField(max_length=255)
+    provider_id = models.CharField(max_length=100, default='EDX')
+
+    class Meta:
+        app_label = 'integrated_channel_plugin_sapsf'
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        return "<OCNWebServicesConfiguration with id {id}>".format(
+            id=self.id
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class OCNWebServicesEnterpriseCustomerConfiguration(TimeStampedModel):
+    """
+    The Enterprise specific configuration we need for integrating with SuccessFactors.
+    """
+
+    enterprise_customer_uuid = models.UUIDField(unique=True)
+    sapsf_base_url = models.CharField(max_length=255)
+    key = models.TextField(blank=True, verbose_name="Client ID")
+    secret = models.TextField(blank=True, verbose_name="Client Secret")
+
+    class Meta:
+        app_label = 'integrated_channel_plugin_sapsf'
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        return "<OCNWebServicesEnterpriseCustomerConfiguration for Enterprise {enterprise_uuid}>".format(
+            enterprise_uuid=self.enterprise_customer_uuid
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class CompletionStatusEventAudit(models.Model):
```

Comment:
If we need audit-trail-like functionality you could review @e-kolpakov's work with django-simple-history in this repo